modules/references/tests/client/ReferencesNew.client.component.tests.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+import ReferencesNew from '../../client/components/ReferencesNew.component';
+import { Self, Loading, Duplicate, Submitted } from '../../client/components/Info';
+import Interaction from '../../client/components/Interaction';
+import Navigation from '../../client/components/Navigation';
+import Enzyme from 'enzyme';
+import { shallow } from 'enzyme';
+import React from 'react';
+import sinon from 'sinon';
+import Adapter from 'enzyme-adapter-react-16';
+import jasmineEnzyme from 'jasmine-enzyme';
+import * as api from '../../client/components/references.api';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+/**
+ * This is a first React test suite with enzyme.
+ * The enzyme configuration can be moved elsewhere (before all tests ever).
+ */
+
+(function () {
+
+  describe('<ReferencesNew />', () => {
+    beforeEach(() => {
+      jasmineEnzyme();
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should not be possible to leave a reference to self', () => {
+      const me = {
+        _id: '123456',
+        username: 'username'
+      };
+
+      const wrapper = shallow(<ReferencesNew userFrom={me} userTo={me} />);
+      expect(wrapper.find(Self)).toExist();
+    });
+
+    it('check whether the reference exists at the beginning', () => {
+      const userFrom = { _id: '111111', username: 'userfrom' };
+      const userTo = { _id: '222222', username: 'userto' };
+      const stub = sinon.stub(api, 'read');
+      stub.withArgs({ userFrom: userFrom._id, userTo: userTo._id }).returns(new Promise(() => {}));
+
+      expect(stub.callCount).toBe(0);
+      const wrapper = shallow(<ReferencesNew userFrom={userFrom} userTo={userTo} />);
+      expect(stub.callCount).toBe(1);
+      expect(wrapper.find(Loading)).toExist();
+    });
+
+    it('can not leave a second reference', async () => {
+      const userFrom = { _id: '111111', username: 'userfrom' };
+      const userTo = { _id: '222222', username: 'userto' };
+      const stub = sinon.stub(api, 'read');
+      stub.withArgs({ userFrom: userFrom._id, userTo: userTo._id }).resolves([{
+        userFrom, userTo, public: false
+      }]);
+
+      const wrapper = shallow(<ReferencesNew userFrom={userFrom} userTo={userTo} />);
+
+      expect(wrapper.find(Duplicate)).not.toExist();
+      await null;
```

Comment:
What do these `await null`s do? 🙃

Reply:
It's waiting for the next tick to resolve the `await api.references.read()` in `componentDidMount`. It's similar to `setTimeout(callback, 0)`. 
I could add a comment or figure out a clearer solution. (Any ideas?)

Reply:
added comments to explain this

Reply:
Thanks, yeah I think the comment is warranted. I didn't know you can `await` pretty much anything. 😁 

Reading about this, folks seem to be using `await 'next tick';` sometimes but `await` + comment is very clear now.

This was an interesting read on topic; https://github.com/nodejs/node/issues/19617